src/main/java/com/example/weatherwaveapi/service/YandexWeatherService.java

```diff
@@ -78,4 +83,22 @@ private YandexApiContainer getYandexApiContainer(String url, String format) {
         }
     }
 
+    private List<CoordinateWeatherRequest> convertCoordinatesRequest(List<String> coordinates) {
+        return Optional.ofNullable(coordinates)
+                .map(this::coordinateBuilder)
+                .orElse(new ArrayList<>());
+    }
+
+    private List<CoordinateWeatherRequest> coordinateBuilder(List<String> coordinates) {
+        return coordinates.stream()
+                .map(coordinate -> coordinate.split(":"))
+                .filter(splitCoordinate -> splitCoordinate.length == 2)
```

Comment:
what happens if there are not 2 coordinates? do we just ignore this situation? we need to handle all possible situations: Check if the list is empty, the record format is invalid, the data format is incorrect. in other words, we need to validate the data and report errors to the user, if it concerns him, so that he can correct his request